### PR TITLE
Load Java class definitions from resources in a safer manner

### PR DESCRIPTION
**Pull request by kelnos**
Makes the source-file-in-resources loading generic (so I can use it easily from other places), and safer (wrap with `Try` and return `Target` instead), and returns a `SupportDefinition` so we can pull import lists from the source file itself.

**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: modules/codegen/src/main/scala/com/twilio/guardrail/generators/syntax/Java.scala [hunk 1]
```diff
@@ -214,13 +215,26 @@ object Java {
...
+                SupportDefinition[JavaLanguage](
+                  new Name(className),
+                  cu.getImports.toList,
+                  clsDef
+                )
+            )
+          )
+    }
+
+  lazy val SHOWER_SUPPORT_DEF: Target[SupportDefinition[JavaLanguage]] = loadSupportDefinitionFromResources("Shower")
```
Comment (blast-hardcheese) [inline modules/codegen/src/main/scala/com/twilio/guardrail/generators/syntax/Java.scala]: Is there any reason to have a singleton here? Why not `def`?

Comment (kelnos) [reply modules/codegen/src/main/scala/com/twilio/guardrail/generators/syntax/Java.scala]: It's intended to be used as an immutable object.  No reason aside from that.

Comment (blast-hardcheese) [reply modules/codegen/src/main/scala/com/twilio/guardrail/generators/syntax/Java.scala]: understood

**Comment by kelnos** [comment]
@blast-hardcheese any objections to merging?

**Pull request merged**